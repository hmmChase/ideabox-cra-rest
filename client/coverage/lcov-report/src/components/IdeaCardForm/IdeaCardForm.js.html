<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/IdeaCardForm/IdeaCardForm.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/IdeaCardForm</a> IdeaCardForm.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>20/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>17/17</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">40x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as thunk from '../../thunks';
import './IdeaCardForm.css';
&nbsp;
export class IdeaCardForm extends Component {
  constructor(props) {
    super(props);
&nbsp;
    this.state = {
      idea: '',
      isSubmitDisabled: true
    };
  }
&nbsp;
  canSubmit = () =&gt; {
    if (this.state.idea === '') {
      this.setState({ isSubmitDisabled: true });
    } else {
      this.setState({ isSubmitDisabled: false });
    }
  };
&nbsp;
  handleChangeIdeaInput = event =&gt; {
    const { name, value } = event.target;
    this.setState({ [name]: value }, () =&gt; this.canSubmit());
  };
&nbsp;
  handleSubmitIdeaForm = async event =&gt; {
    event.preventDefault();
    this.setState({ isSubmitDisabled: true });
    await this.props.dispatchAddIdea(this.state.idea);
    this.setState({ idea: '' }, () =&gt; this.canSubmit());
  };
&nbsp;
  render() {
    return (
      &lt;form
        className="IdeaCardForm"
        onSubmit={event =&gt; this.handleSubmitIdeaForm(event)}
      &gt;
        &lt;input
          className="IdeaCardInput"
          name="idea"
          type="text"
          placeholder="Enter idea"
          value={this.state.idea}
          onChange={event =&gt; this.handleChangeIdeaInput(event)}
        /&gt;
        &lt;button type="submit" disabled={this.state.isSubmitDisabled}&gt;
          Add idea
        &lt;/button&gt;
      &lt;/form&gt;
    );
  }
}
&nbsp;
export const mapDispatchToProps = dispatch =&gt; ({
  dispatchAddIdea: idea =&gt; dispatch(thunk.addIdea(idea))
});
&nbsp;
IdeaCardForm.propTypes = {
  dispatchAddIdea: PropTypes.func.isRequired
};
&nbsp;
export default connect(
  null,
  mapDispatchToProps
)(IdeaCardForm);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 09 2018 06:33:58 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
