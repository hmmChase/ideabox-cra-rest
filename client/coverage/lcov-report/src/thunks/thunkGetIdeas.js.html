<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/thunks/thunkGetIdeas.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/thunks</a> thunkGetIdeas.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as action from '../actions';
import * as api from '../api';
&nbsp;
export const getIdeas = url =&gt; {
  return async dispatch =&gt; {
    try {
      dispatch(action.hasErrored(false));
      dispatch(action.isLoading(true));
      const ideas = await api.request(url);
      dispatch(action.setIdeas(ideas));
      dispatch(action.isLoading(false));
    } catch (error) {
      dispatch(action.isLoading(false));
      dispatch(action.hasErrored(true));
    }
  };
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 09 2018 06:33:58 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
