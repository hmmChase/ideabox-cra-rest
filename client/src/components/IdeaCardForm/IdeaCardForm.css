form {
position: relative;display: flex;
width: 80%;
max-width: 800px;
}

img {
  position: absolute;
  top: -60px;
  left: -45px;
  width: 100px;
}

textarea {
  width: 100%;
  padding: 10px;
  padding-left: 20px;
  outline: none;
  resize: vertical;
}

button {
  width: 60px;
  border: none;
  background-color: #fffacd;
}
