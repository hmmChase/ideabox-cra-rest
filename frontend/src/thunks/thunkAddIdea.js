import * as action from '../actions';
import * as api from '../api';

export const addIdea = idea => async dispatch => {
  try {
    dispatch(action.hasErrored(false));
    const responseIdea = await api.postIdea(idea);
    dispatch(action.setIdea(responseIdea));
  } catch (error) {
    dispatch(action.hasErrored(true));
  }
};
