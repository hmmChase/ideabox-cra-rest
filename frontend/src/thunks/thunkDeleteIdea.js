import * as action from '../actions';
import * as api from '../api';

export const deleteIdea = (ideaID, ideas) => {
  return async dispatch => {
    try {
      dispatch(action.hasErrored(false));
      await api.deleteIdea(ideaID);
      const filteredIdeas = ideas.filter(idea => {
        return idea.id !== ideaID;
      });
      dispatch(action.setIdeas(filteredIdeas));
    } catch (error) {
      dispatch(action.hasErrored(true));
    }
  };
};
