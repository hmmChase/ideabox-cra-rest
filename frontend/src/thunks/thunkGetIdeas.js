import * as action from '../actions';
import * as api from '../api';

export const getIdeas = url => {
  return async dispatch => {
    try {
      dispatch(action.hasErrored(false));
      dispatch(action.isLoading(true));
      const ideas = await api.request(url);
      dispatch(action.setIdeas(ideas));
      dispatch(action.isLoading(false));
    } catch (error) {
      dispatch(action.isLoading(false));
      dispatch(action.hasErrored(true));
    }
  };
};
