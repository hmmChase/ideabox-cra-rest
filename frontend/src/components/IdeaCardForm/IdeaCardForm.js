import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as thunk from '../../thunks';
import './IdeaCardForm.css';
import ideaboxImg from '../../assets/images/ideabox.png';

export class IdeaCardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      idea: '',
      isSubmitDisabled: true
    };
  }

  canSubmit = () => {
    if (this.state.idea === '') {
      this.setState({ isSubmitDisabled: true });
    } else {
      this.setState({ isSubmitDisabled: false });
    }
  };

  handleChangeIdeaInput = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value }, () => this.canSubmit());
  };

  handleSubmitIdeaForm = async event => {
    event.preventDefault();
    this.setState({ isSubmitDisabled: true });
    await this.props.dispatchAddIdea(this.state.idea);
    this.setState({ idea: '' }, () => this.canSubmit());
  };

  render() {
    return (
      <form
        className="IdeaCardForm"
        onSubmit={event => this.handleSubmitIdeaForm(event)}>
        <img src={ideaboxImg} alt="ideabox" />
        <textarea
          className="IdeaCardInput"
          name="idea"
          type="text"
          placeholder="What's on your mind?"
          value={this.state.idea}
          onChange={event => this.handleChangeIdeaInput(event)}
        />
        <button type="submit" disabled={this.state.isSubmitDisabled}>
          Add Idea
        </button>
      </form>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  dispatchAddIdea: idea => dispatch(thunk.addIdea(idea))
});

IdeaCardForm.propTypes = {
  dispatchAddIdea: PropTypes.func.isRequired
};

export default connect(
  null,
  mapDispatchToProps
)(IdeaCardForm);
