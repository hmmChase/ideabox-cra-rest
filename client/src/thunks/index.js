import { getIdeas } from './thunkGetIdeas';
import { addIdea } from './thunkAddIdea';
import { updateIdea } from './thunkUpdateIdea';
import { deleteIdea } from './thunkDeleteIdea';

export { getIdeas, addIdea, updateIdea, deleteIdea };
