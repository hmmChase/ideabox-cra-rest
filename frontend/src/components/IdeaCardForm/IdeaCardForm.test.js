import React from 'react';
import { shallow } from 'enzyme';
import { IdeaCardForm, mapDispatchToProps } from './IdeaCardForm';
import * as thunk from '../../thunks';
jest.mock('../../thunks');

describe('IdeaCardForm', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {
      dispatchAddIdea: jest.fn()
    };

    wrapper = shallow(<IdeaCardForm {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches the snapshot with idea', () => {
    wrapper.setState({ idea: 'mock idea' });

    expect(wrapper).toMatchSnapshot();
  });

  it('matches the snapshot with isSubmitDisabled: false', () => {
    wrapper.setState({ isSubmitDisabled: false });

    expect(wrapper).toMatchSnapshot();
  });

  describe('canSubmit', () => {
    it('sets state.isSubmitDisabled: true if input is empty', () => {
      wrapper.setState({ idea: '', isSubmitDisabled: false });

      wrapper.instance().canSubmit();

      expect(wrapper.state().isSubmitDisabled).toBe(true);
    });

    it('sets state.isSubmitDisabled: false if input isnt empty', () => {
      wrapper.setState({ idea: 'mock idea', isSubmitDisabled: true });

      wrapper.instance().canSubmit();

      expect(wrapper.state().isSubmitDisabled).toBe(false);
    });
  });

  describe('handleChangeIdeaInput', () => {
    it('sets state with tag name and value', () => {
      const event = { target: { name: 'idea', value: 'mock idea' } };

      expect(wrapper.state().idea).toEqual('');

      wrapper.instance().handleChangeIdeaInput(event);

      expect(wrapper.state().idea).toEqual('mock idea');
    });

    it('calls canSubmit', () => {
      const event = { target: { name: 'idea', value: 'mock idea' } };
      const canSubmit = (wrapper.instance().canSubmit = jest.fn());

      wrapper.instance().handleChangeIdeaInput(event);

      expect(canSubmit).toHaveBeenCalledTimes(1);
    });
  });

  describe('handleSubmitIdeaForm', () => {
    it('calls preventDefault', async () => {
      const event = { preventDefault: jest.fn() };

      await wrapper.instance().handleSubmitIdeaForm(event);

      expect(event.preventDefault).toHaveBeenCalledTimes(1);
    });

    it('sets state.isSubmitDisabled: true', async () => {
      const event = { preventDefault: jest.fn() };
      wrapper.setState({ isSubmitDisabled: false });

      expect(wrapper.state().isSubmitDisabled).toBe(false);

      await wrapper.instance().handleSubmitIdeaForm(event);

      expect(wrapper.state().isSubmitDisabled).toBe(true);
    });

    it('calls dispatchAddIdea(this.state.idea)', async () => {
      const event = { preventDefault: jest.fn() };

      await wrapper.instance().handleSubmitIdeaForm(event);

      expect(mockProps.dispatchAddIdea).toHaveBeenCalledTimes(1);
    });

    it('sets state.idea to empty string', async () => {
      const event = { preventDefault: jest.fn() };
      wrapper.setState({ idea: 'mock idea' });

      expect(wrapper.state().idea).toEqual('mock idea');

      await wrapper.instance().handleSubmitIdeaForm(event);

      expect(wrapper.state().idea).toEqual('');
    });

    it('calls canSubmit', async () => {
      const event = { preventDefault: jest.fn() };
      const canSubmit = (wrapper.instance().canSubmit = jest.fn());

      await wrapper.instance().handleSubmitIdeaForm(event);

      expect(canSubmit).toHaveBeenCalledTimes(1);
    });
  });

  describe('event handlers', () => {
    it('calls handleSubmitIdeaForm(event) on IdeaCardForm submit', () => {
      const event = {};
      const handleSubmitIdeaForm = (wrapper.instance().handleSubmitIdeaForm = jest.fn());

      wrapper.find('.IdeaCardForm').simulate('submit', event);

      expect(handleSubmitIdeaForm).toHaveBeenCalledTimes(1);
      expect(handleSubmitIdeaForm).toHaveBeenCalledWith(event);
    });

    it('calls handleChangeIdeaInput(event) on IdeaCardInput change', () => {
      const event = {};
      const handleChangeIdeaInput = (wrapper.instance().handleChangeIdeaInput = jest.fn());

      wrapper.find('.IdeaCardInput').simulate('change', event);

      expect(handleChangeIdeaInput).toHaveBeenCalledTimes(1);
      expect(handleChangeIdeaInput).toHaveBeenCalledWith(event);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch on addIdea(idea)', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);

      const mockIdea = { idea: 'mock idea' };

      mappedProps.dispatchAddIdea(mockIdea);

      expect(mockDispatch).toHaveBeenCalledTimes(1);
      expect(thunk.addIdea).toHaveBeenCalledTimes(1);
      expect(thunk.addIdea).toHaveBeenCalledWith(mockIdea);
    });
  });
});
