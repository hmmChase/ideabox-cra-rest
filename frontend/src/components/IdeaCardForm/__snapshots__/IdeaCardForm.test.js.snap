// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCardForm matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaCardForm
    dispatchAddIdea={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="IdeaCardInput"
          name="idea"
          onChange={[Function]}
          placeholder="Enter idea"
          type="text"
          value=""
        />,
        <button
          disabled={true}
          type="submit"
        >
          Add idea
        </button>,
      ],
      "className": "IdeaCardForm",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "IdeaCardInput",
          "name": "idea",
          "onChange": [Function],
          "placeholder": "Enter idea",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add idea",
          "disabled": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Add idea",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="IdeaCardInput"
            name="idea"
            onChange={[Function]}
            placeholder="Enter idea"
            type="text"
            value=""
          />,
          <button
            disabled={true}
            type="submit"
          >
            Add idea
          </button>,
        ],
        "className": "IdeaCardForm",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "IdeaCardInput",
            "name": "idea",
            "onChange": [Function],
            "placeholder": "Enter idea",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add idea",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Add idea",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`IdeaCardForm matches the snapshot with idea 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaCardForm
    dispatchAddIdea={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="IdeaCardInput"
          name="idea"
          onChange={[Function]}
          placeholder="Enter idea"
          type="text"
          value="mock idea"
        />,
        <button
          disabled={true}
          type="submit"
        >
          Add idea
        </button>,
      ],
      "className": "IdeaCardForm",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "IdeaCardInput",
          "name": "idea",
          "onChange": [Function],
          "placeholder": "Enter idea",
          "type": "text",
          "value": "mock idea",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add idea",
          "disabled": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Add idea",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="IdeaCardInput"
            name="idea"
            onChange={[Function]}
            placeholder="Enter idea"
            type="text"
            value="mock idea"
          />,
          <button
            disabled={true}
            type="submit"
          >
            Add idea
          </button>,
        ],
        "className": "IdeaCardForm",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "IdeaCardInput",
            "name": "idea",
            "onChange": [Function],
            "placeholder": "Enter idea",
            "type": "text",
            "value": "mock idea",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add idea",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Add idea",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`IdeaCardForm matches the snapshot with isSubmitDisabled: false 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaCardForm
    dispatchAddIdea={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="IdeaCardInput"
          name="idea"
          onChange={[Function]}
          placeholder="Enter idea"
          type="text"
          value=""
        />,
        <button
          disabled={false}
          type="submit"
        >
          Add idea
        </button>,
      ],
      "className": "IdeaCardForm",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "IdeaCardInput",
          "name": "idea",
          "onChange": [Function],
          "placeholder": "Enter idea",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add idea",
          "disabled": false,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Add idea",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="IdeaCardInput"
            name="idea"
            onChange={[Function]}
            placeholder="Enter idea"
            type="text"
            value=""
          />,
          <button
            disabled={false}
            type="submit"
          >
            Add idea
          </button>,
        ],
        "className": "IdeaCardForm",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "IdeaCardInput",
            "name": "idea",
            "onChange": [Function],
            "placeholder": "Enter idea",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add idea",
            "disabled": false,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Add idea",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
